src/KnownProperties.jl

```diff
@@ -0,0 +1,67 @@
+########################################################################
+# This is to introduce a common language to ask whether or not a given
+# property (which is potentially hard to compute) is known for some
+# object `x`. 
+########################################################################
+
+@doc raw"""
+    is_known(x::Any, f::Function)
+
+Return whether a property of an object `x` called for by `f(x)` is known. 
+
+Note: The default implementation throws an error. It is the programmer's 
+responsibility to implement appropriate methods for their individual 
+types and properties. See `src/KnownProperties.jl` for details.
+"""
+function is_known(x::Any, f::Function)
+  # If the object `x` has attributes and an attribute with the name 
+  # of `f` happens to be stored, return it in good faith.
+  hasfield(typeof(x), :__attrs) && has_attribute(x, nameof(f)) && return true
```

Comment:
I don't like that the docstring states that the default implementation gives an error, but then there is an implementation here.
Furthermore, IMO, users should never need to handle attributes. It is an implementation detail, how the attribute is named; it needn't be the same as the corresponding getter function, what this implementation here expects.